packages/table/src/Table.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useReducer, ReactElement } from 'react';
+import { cx, css } from '@leafygreen-ui/emotion';
+import { HeaderRowProps } from './HeaderRow';
+import { TableHeaderProps } from './TableHeader';
+import { State, Types, TableProvider, reducer } from './table-context';
+import TableHead from './TableHead';
+
+// * fix row appearing as children of row -- not seeing this!
+
+const tableStyles = css`
+  border-collapse: collapse;
+  box-sizing: border-box;
+`;
+
+type DataShape<Shape> = Shape extends infer U ? U : Shape;
+
+interface TableRowInterface<Shape> {
+  datum: DataShape<Shape>;
+  index?: number;
+}
+
+export interface TableProps<Shape>
+  extends React.ComponentPropsWithoutRef<'table'> {
+  data: Array<DataShape<Shape>>;
+  columns:
+    | Array<ReactElement<HeaderRowProps>>
+    | Array<ReactElement<TableHeaderProps> | string>
+    | React.ReactFragment;
+  selectable?: boolean;
+  children: (TableRowArgs: TableRowInterface<Shape>) => JSX.Element;
+}
+
+export default function Table<Shape>({
+  columns = [],
+  data = [],
+  selectable: selectableProp = false,
+  children,
+  className,
+  ...rest
+}: TableProps<Shape>) {
+  const initialState: State = {
+    sort: {
+      direction: 'asc',
+    },
+    data,
+    selectable: selectableProp,
+    headerCheckState: false,
+    headerIndeterminate: false,
+    hasNestedRows: false,
+  };
+
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  React.useEffect(() => {
```

Comment:
Let's be consistent with how we use hooks. We have a plain `useReducer` above and then this is off the React object. I prefer importing the individual functions, but prefer consistency above all else.

Reply:
Sorry yes good point

Reply:
👍